overlord/snapstate/snapmgr.go

```diff
@@ -884,8 +883,13 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	if cand.Revision.Local() {
+		snapst.LocalRevision = ss.Revision()
+	}
+
```

Comment:
sorry, sorry, now I understand what is going on, it looks correct, but there is various code that can go away, see my general comment about looking at each Set(st, ss.Name(), snapst) and snapst change and remove the now pointless ones